repose-aggregator/components/filters/url-extractor-to-header/src/main/scala/org/openrepose/filters/urlextractortoheader/UrlExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.urlextractortoheader
+
+import java.net.URL
+import javax.inject.{Named, Inject}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.urlextractortoheader.config.UrlExtractorToHeaderConfig
+
+import scala.collection.JavaConverters._
+import scala.util.matching.Regex
+
+@Named
+class UrlExtractorToHeaderFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[UrlExtractorToHeaderConfig] with LazyLogging {
+
+  import UrlExtractorToHeaderFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var extractions: Iterable[Extraction] = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
```

Comment:
We should extract this to a base class, because it looks pretty much exactly the same for every filter we write. This is not work for you right now, i'm just making a note.